xwiki-platform-core/xwiki-platform-livedata/xwiki-platform-livedata-webjar/src/main/webjar/Logic.js

```diff
@@ -59,520 +67,763 @@ define(["jquery", "polyfills"], function ($) {
+     * @param {String} propertyId
+     * @returns {Object}
+     */
+    getPropertyDescriptor: function (propertyId) {
+      return this.data.meta.propertyDescriptors.find(function (propertyDescriptor) {
+        return propertyDescriptor.id === propertyId;
+      });
+    },
+
+
+    /**
+     * Return the property type descriptor corresponding to a property id
+     * @param {String} propertyId
+     * @returns {Object}
+     */
+    getPropertyTypeDescriptor: function (propertyId) {
+      var propertyDescriptor = this.getPropertyDescriptor(propertyId);
+      return this.data.meta.propertyTypes.find(function (typeDescriptor) {
+        return typeDescriptor.id === propertyDescriptor.type;
+      });
+    },
+
+
+    /**
+     * Return the property type descriptor corresponding to a property id
+     * @param {String} propertyId
+     * @returns {Object}
+     */
+    getLayoutDescriptor: function (layoutId) {
+      return this.data.meta.layoutDescriptors.find(function (layoutDescriptor) {
+        return layoutDescriptor.id === layoutId;
+      });
+    },
+
+
+    /**
+     * Get the displayer descriptor associated to a property id
+     * @param {String} propertyId
+     */
+    getDisplayerDescriptor: function (propertyId) {
+      if (!this.isValidPropertyId(propertyId)) { return; }
+      // property descriptor config
+      var propertyDescriptor = this.getPropertyDescriptor(propertyId);
+      var propertyDescriptorDisplayer = propertyDescriptor.displayer || {};
+      // property type descriptor config
+      var typeDescriptor = this.getPropertyTypeDescriptor(propertyId);
+      var typeDescriptorDisplayer = typeDescriptor.displayer || {};
+      // default displayer config
+      var displayerId = propertyDescriptorDisplayer.id || typeDescriptorDisplayer.id;
+      var displayer = this.data.meta.displayers.find(function (displayer) {
+        return displayer.id === displayerId;
+      });
+
+      return $.extend({}, displayer, typeDescriptorDisplayer, propertyDescriptorDisplayer);
+    },
+
+
+    /**
+     * Get the filter descriptor associated to a property id
+     * @param {String} propertyId
+     */
+    getFilterDescriptor: function (propertyId) {
+      if (!this.isValidPropertyId(propertyId)) { return; }
+      // property descriptor config
+      var propertyDescriptor = this.getPropertyDescriptor(propertyId);
+      var propertyDescriptorFilter = propertyDescriptor.filter || {};
+      // property type descriptor config
+      var typeDescriptor = this.getPropertyTypeDescriptor(propertyId);
+      var typeDescriptorFilter = typeDescriptor.filter || {};
+      // default filter config
+      var filterId = propertyDescriptorFilter.id || typeDescriptorFilter.id;
+      var filter = this.data.meta.filters.find(function (filter) {
+        return filter.id === filterId;
+      });
+
+      return $.extend({}, filter, typeDescriptorFilter, propertyDescriptorFilter);
```

Comment:
> for displayers we fall back on the default displayer if none is specified (neither in the property descriptor nor in the type descriptor) because we want (?) to display the property even if we don't have any information about the displayer

Indeed, I think the data should always be displayed even if we don't have any information about the displayer, as there is a property descriptor, that mean that this property was meant to be there.

If we want the displayers and the filters to have a similar behavior, we could do as we do for the filter:
- filter/displayer object with null id: un-filterable/displayable (do not display filter/displayer)
- no filter/displayer object: merge with corresponding typeDescriptor.filter/displayer descriptor

But having an undisplayable property sound too weird for me!